lib/ansible/modules/network/nuage/nuage_vspk.py

```diff
@@ -0,0 +1,1089 @@
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.auth = module.params['auth']
+        self.api_username = None
+        self.api_password = None
+        self.api_enterprise = None
+        self.api_url = None
+        self.api_version = None
+        self.api_certificate = None
+        self.api_key = None
+        self.type = module.params['type']
+
+        self.state = None
+        if 'state' in list(module.params.keys()):
+            self.state = module.params['state']
+
+        self.command = None
+        if 'command' in list(module.params.keys()):
+            self.command = module.params['command']
+
+        self.match_filter = None
+        if 'match_filter' in list(module.params.keys()):
+            self.match_filter = module.params['match_filter']
+
+        self.entity_id = None
+        if 'id' in list(module.params.keys()):
+            self.entity_id = module.params['id']
+
+        self.parent_id = None
+        if 'parent_id' in list(module.params.keys()):
+            self.parent_id = module.params['parent_id']
+
+        self.parent_type = None
+        if 'parent_type' in list(module.params.keys()):
+            self.parent_type = module.params['parent_type']
+
+        self.properties = None
+        if 'properties' in list(module.params.keys()):
+            self.properties = module.params['properties']
+
+        self.children = None
+        if 'children' in list(module.params.keys()):
+            self.children = module.params['children']
+
+        self.entity = None
+        self.entity_class = None
+        self.parent = None
+        self.parent_class = None
+        self.entity_fetcher = None
+
+        self.result = {
+            'state': self.state,
+            'id': self.entity_id,
+            'entities': []
+        }
+        self.nuage_connection = None
+
+        self._verify_api()
+        self._verify_input()
+        self._connect_vspk()
+        self._find_parent()
+
+    def _connect_vspk(self):
+        """
+        Connects to a Nuage API endpoint
+        """
+        try:
+            # Connecting to Nuage
+            if self.api_certificate and self.api_key:
+                self.nuage_connection = VSPK.NUVSDSession(username=self.api_username, enterprise=self.api_enterprise, api_url=self.api_url,
+                                                          certificate=(self.api_certificate, self.api_key))
+            else:
+                self.nuage_connection = VSPK.NUVSDSession(username=self.api_username, password=self.api_password, enterprise=self.api_enterprise,
+                                                          api_url=self.api_url)
+            self.nuage_connection.start()
+        except BambouHTTPError as error:
+            self.module.fail_json(msg='Unable to connect to the API URL with given username, password and enterprise: {0}'.format(error))
+
+    def _verify_api(self):
```

Comment:
This is something we are planning once we move to more modules, we'll split the general functionalities (not only this, but building the connection as well) into a module_util. Currently this is not needed yet, i think.